algorl/src/TD.py

```diff
@@ -530,4 +539,113 @@ def compute_state_values(self):
             self.env.grid[grid_state]=state_values[state_values.states == grid_state].state_value
                 
 
-        
\ No newline at end of file
+class Sarsa_Lambda(TemporalDifferenceFunctions):
+    """ 
+    SARSA Algorithm that utilizes the lambda target instead of the TD target.
+    Trace vector can be set as either accumulating or replacing. 
+    Policy is epsilon-greedy. 
+    Reference:
+    --------------------
+    - Grokking Deep Reinforcement Learning by Miguel Morales. Page 210.
+    """
+
+    def __init__(
+            self, env, alpha:float = 0.5, gamma:float = 0.9, lambda_:float = 0.5, 
+            epsilon:bool = 0.1, replacing_traces:bool = False, starting_state = (2,0), 
+            num_of_epochs:int = 1_000, num_episodes = 10_000, plot_name = 'Sarsa_Lambda', 
+            reward = -1):
```

Comment:
Same here, int, str and int, missing.
You can also try to add the expected result(s) using '->'
check the decumentation